### PR TITLE
unit-test: Add unit-test for CloneAPI Template-Filters

### PR DESCRIPTION
**Pull request by matthewei**
**What this PR does / why we need it**:
This commit is for CloneAPI unit-test. [The CloneAPI add a template-filter function](https://kubevirt.io/user-guide/operations/clone_api/#:~:text=as%20a%20default.-,Template%20Label%20%26%20Template%20Annotation%20filters,-%C2%B6) so I want to add the template-filter unit-test for that.
Fixes #

**Special notes for your reviewer**:

**Checklist**

This checklist is not enforcing, but it's a reminder of items that could be relevant to every PR.
Approvers are expected to review this list.

- [ ] Design: A [design document](https://github.com/kubevirt/community/tree/main/design-proposals) was considered and is present (link) or not required
- [ ] PR: The PR description is expressive enough and will help future contributors
- [ ] Code: [Write code that humans can understand](https://en.wikiquote.org/wiki/Martin_Fowler#code-for-humans) and [Keep it simple](https://en.wikipedia.org/wiki/KISS_principle)
- [ ] Refactor: You have [left the code cleaner than you found it (Boy Scout Rule)](https://learning.oreilly.com/library/view/97-things-every/9780596809515/ch08.html)
- [ ] Upgrade: Impact of this change on upgrade flows was considered and addressed if required
- [ ] Testing: New code requires [new unit tests](https://github.com/kubevirt/kubevirt/blob/main/docs/reviewer-guide.md#when-is-a-pr-good-enough). New features and bug fixes require at least on e2e test
- [ ] Documentation: A [user-guide update](https://github.com/kubevirt/user-guide/) was considered and is present (link) or not required. You want a user-guide update if it's a user facing feature / API change.
- [ ] Community: Announcement to [kubevirt-dev](https://groups.google.com/g/kubevirt-dev/) was considered

**Release note**:

```release-note
None
```

### CODE REVIEW

File: pkg/virt-controller/watch/clone/clone_test.go [hunk 1]
```diff
@@ -873,14 +873,21 @@ var _ = Describe("Clone", func() {
...
-				if labelOrAnnotation == labels {
+				switch labelOrAnnotation {
+				case labels:
 					expectedVM.Labels = m
-				} else {
+				case annotations:
 					expectedVM.Annotations = m
+				case templateLabels:
+					expectedVM.Spec.Template.ObjectMeta.Labels = m
+				case templateAnnotations:
```
Comment (fossedihelm) [inline pkg/virt-controller/watch/clone/clone_test.go:889]: please add a failing `default` case :)

**Review by fossedihelm — COMMENTED** [review]
Thank you! Just a nit below
/test all

File: pkg/virt-controller/watch/clone/clone_test.go (hunk 1 above)
Comment (matthewei) [reply pkg/virt-controller/watch/clone/clone_test.go:889]: yes, will do it! Many thanks!

Comment (matthewei) [reply pkg/virt-controller/watch/clone/clone_test.go:889]: done!

**Review by fossedihelm — APPROVED** [review]
Thank you!

**Pull request closed without merging**